egraph.rkt

```diff
@@ -133,80 +172,108 @@
 
 (define (dedup-vars! en)
   (update-vars! (pack-leader en) update-en-expr)
-  (dedup-children! en))
+  (dedup-children! (pack-leader en)))
```

Comment:
**Bug 2**
We called dedup-vars! in merge-egraph-nodes sometimes without referring to the parent, which was bad because about half the time this call did nothing and took a long time doing it.